addons/website_sale/static/src/js/website_sale.js

```diff
@@ -193,6 +193,7 @@ $('.oe_website_sale').each(function () {
             $product_id.val(0);
             $parent.find(".js_check_product").attr("disabled", "disabled");
         }
+        $product_id.trigger('change');
```

Comment:
Actually, setting the value of the input with javascript will not trigger automatically the change ? Do you really need to trigger it manually ?